pydropsonde/circles.py

```diff
@@ -18,3 +24,102 @@ class Circle:
     def dummy_circle_function(self):
         print(self.flight_id, self.segment_id)
         return self
+
+    def get_div_and_vor(self):
+        D = self.circle_ds.dudx + self.circle_ds.dvdy
+        vor = self.circle_ds.dvdx - self.circle_ds.dudy
+
+        vor_attrs = {
+            "standard_name": "atmosphere_relative_vorticity",
+            "long_name": "Area-averaged horizontal relative vorticity",
+            "units": str(1 / units.second),
+        }
+        D_attrs = {
+            "standard_name": "divergence_of_wind",
+            "long_name": "Area-averaged horizontal mass divergence",
+            "units": str(1 / units.second),
+        }
+        self.circle_ds = self.circle_ds.assign(
+            dict(D=(["alt"], D.values, D_attrs), vor=(["alt"], vor.values, vor_attrs))
+        )
+        return self
+
+    def get_density(self, sonde_dim="sonde_id"):
+        mr = mpcalc.mixing_ratio_from_specific_humidity(
+            self.circle_ds.q.values,
+        )
+        density = mpcalc.density(
+            self.circle_ds.p.values * units.Pa,
+            self.circle_ds.ta.values * units.kelvin,
+            mr,
+        )
+        density_attrs = {
+            "standard_name": "air_density",
+            "long_name": "Air density",
+            "units": str(density.units),
+        }
+        mean_density_attrs = {
+            "standard_name": "mean_air_density",
+            "long_name": "Mean air density in circle",
+            "units": str(density.units),
+        }
+        self.circle_ds = self.circle_ds.assign(
+            dict(
+                density=(self.circle_ds.ta.dims, density.magnitude, density_attrs),
+                mean_density=(
+                    ["alt"],
+                    self.circle_ds.density.mean(sonde_dim),
+                    mean_density_attrs,
+                ),
+            )
+        )
+
+        return self
+
+    def get_vertical_velocity(self):
+        div = self.circle_ds.div.where(~np.isnan(self.circle_ds.div), drop=True).sortby(
+            "alt"
+        )
+        zero_vel = xr.DataArray(data=[0], dims="alt", coords={"alt": [0]})
+
+        height = xr.concat([zero_vel, div["alt"]], dim="alt")
+        height_diff = height.diff(dim="alt")
+
+        del_w = -div * height_diff.values
```

Comment:
Hi all, wow everything seems to have advanced so much already ! Not sure to where leave my comment but there is a problem in this definition of vertical velocity because it assumes air is incompressible. Instead omega should be computed first by integrating divergence over pressure (which accounts for air compressibility), and then vertical velocity can be computed by dividing by rho*g. Is that clear ? I can update those two functions if you want as I have already done it for the JOANNE dataset, but I am very unfamiliar with GitHub and don't know where to do this

Reply:
Hi @bpoujol you are the best person to help out with this indeed! :) I think the [contributing section](https://atmdrops.github.io/pydropsonde/CONTRIBUTING.html) might help you get started on how you can contribute, but if you have any questions you can always ask here.

Reply:
Oops, just noticed some links need to be updated in there (#58). but this should be done rather quickly.

Reply:
Thanks so much for chimming in @bpoujol 
Geet already pointed us to you PR for JOANNE and I am sure, @ninarobbins will change this functions accordingly. 
Another way would be for you to make a PR to her repo or we could remove omega and vertical velocity from this PR and make another one for them.

Reply:
Thank you Geet for the documentation, very useful ! For Helene and Nina : it is as you want. If Nina plans to do it then it is great ! But the function in my JOANNE PR looks quite complicated and I am sure it is possible to make it simpler.    Else, if you prefer, I can either do a PR to the repo of @ninarobbins or I wait for this PR to be incorporated in the main code before I do a second PR. Let me know what is the best !

Reply:
Dear @bpoujol, sorry for the super late reply, I just got back to working on this. I am happy if you do a PR to my repository, as I would prefer not to incorporate this when it is wrong. Would you still be able to do so?

Reply:
Hi @ninarobbins , sorry for the late reply as well. I am very busy right now and probably still for the coming one or two weeks, is it fine if I only do this afterwards ?